docs/branching.md

```diff
@@ -0,0 +1,98 @@
+# Branching and deployments model
+
+This document aims to discuss and document the relations between branches and deployments of `elixir-omg`, with respect to branches and deployments of `plasma-contracts`.
+
+It is a refinement of the [OIP4 branching model](https://github.com/omisego/OIP/blob/master/0004-ewallet-release-and-versioning.md), applicable to `elixir-omg` and `plasma-contracts` versioning.
+
+Rationale:
+- the history and the relations between the versions are readable and simple to understand
+- we can predictably sync our respective watchers/run child chain servers against deployed contracts
+- we can move with the `master` branch quickly
+
+
+## Dependency Rules for elixir-omg
+
+### For mix.exs
+- `elixir-omg/master` will point to the `plasma-contracts/master`
+- A release branch in `elixir-omg` will point to the corresponding release branch in `plasma-contracts`. For example:
+  - `elixir-omg/v0.1` -> `plasma-contracts/v0.1`
+
+### For mix.lock
+- Always points to a specific SHA that in the history of the `plasma-contracts` branch referenced in `mix.exs`
+
+
+## Deployment Scenarios
+
```

Comment:
It might be worthwhile to add a summary difference for each of our environments or linking into another document for further reading.
From [some internal documentation](https://github.com/omisego/devops/wiki/DevOps-Wiki), we could do a brief description like:

```suggestion

We have 3 types of environments that we deploy the project into:

### Development environment

Development environment is for builds that land in the master branch. This environment should be considered unstable as all commits to master will be deployed here. Continuous deployment is configured for this cluster. There are two deployment instances:

* One set of services connected to Rinkeby with Geth
* one connected to Görli with Parity.

### Staging environment

This environment should be treated as an always up environment as we will have integration from the other services that require blockchain service interaction. Continuous deployment is configured for this cluster.
```

The above is a partial suggestion and could be detailed further.

Reply:
Our deployment details are kept in our devops documentation. Will detail there.